src/transformState.js

```diff
@@ -5,7 +5,39 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
```

Comment:
How about using this syntax to destructure an object here, instead of every case of switch.

```suggestion
  for (const { type, extraData, keysToRemove } of actions) {
```